include/uacpi/acpi.h

```diff
@@ -1003,6 +1003,66 @@ UACPI_PACKED(struct acpi_ecdt {
 })
 UACPI_EXPECT_SIZEOF(struct acpi_ecdt, 65);
 
+#define ACPI_DBG2_TYPE_SERIAL 0x8000
+#define ACPI_DBG2_TYPE_1394 0x8001
+#define ACPI_DBG2_TYPE_USB 0x8002
+#define ACPI_DBG2_TYPE_NET 0x8003
+
+#define ACPI_DBG2_SUBTYPE_SERIAL_NS16550 0x0
+#define ACPI_DBG2_SUBTYPE_SERIAL_NS16550_DBGP1 0x1
+#define ACPI_DBG2_SUBTYPE_SERIAL_PL011 0x3
+
+UACPI_PACKED(struct acpi_dbg2 {
+    struct acpi_sdt_hdr hdr;
+    uacpi_u32 offset_dbg_device_info;
+    uacpi_u32 number_dbg_device_info;
+})
+UACPI_EXPECT_SIZEOF(struct acpi_dbg2, 44);
+
+UACPI_PACKED(struct acpi_dbg2_dbg_device_info {
+    uacpi_u8 revision;
+    uacpi_u16 length;
+    uacpi_u8 number_generic_address_registers;
+    uacpi_u16 namespace_string_length;
+    uacpi_u16 namespace_string_offset;
+    uacpi_u16 oem_data_length;
+    uacpi_u16 oem_data_offset;
+    uacpi_u16 port_type;
+    uacpi_u16 port_subtype;
+    uacpi_u16 reserved;
+    uacpi_u16 base_address_register_offset;
+    uacpi_u16 address_size_offset;
+})
+UACPI_EXPECT_SIZEOF(struct acpi_dbg2_dbg_device_info, 22);
+
+UACPI_PACKED(struct acpi_spcr {
+    struct acpi_sdt_hdr hdr;
+    uacpi_u8 interface_type;
```

Comment:
An enum would be nice

Reply:
This is identical to the DBG2 subtype field (and there is no type field -- SPCR always refers to a serial device). I'll add a comment to clarify this.